cpp/src/io/csv/reader_impl.cu

```diff
@@ -419,11 +419,13 @@ std::pair<rmm::device_uvector<char>, selected_rows_offsets> select_data_and_row_
       reinterpret_cast<const char*>(buffer->data()),
       buffer->size());
 
-    std::vector<char> h_uncomp_data_owner;
+    std::vector<uint8_t> h_uncomp_data_owner;
 
     if (reader_opts.get_compression() != compression_type::NONE) {
-      h_uncomp_data_owner = get_uncompressed_data(h_data, reader_opts.get_compression());
-      h_data              = h_uncomp_data_owner;
+      h_uncomp_data_owner =
+        decompress(reader_opts.get_compression(), {buffer->data(), buffer->size()});
```

Comment:
You have `h_data` built above that is a `host_span`, why construct another one when you can use that? Do you have a type conflict with it?

Reply:
I'm not sure what you mean. Before this branch, `h_data` is a view of the raw input data. If there's compression, we assign a view of the decompressed data (`h_uncomp_data_owner`) to the same `host_span`.

Reply:
The call to `decompress` takes a host span as a second argument. I was wondering why we couldn't simply pass h_data instead of constructing a new host span. Based on other comments, it could be type related though where decompress takes a span of uint8_t's. If that is the case, feel free to resolve this comment.

Reply:
yeah, decompress takes a `uint8_t` span. Starting to think this is something to be improved (allow more input types).